internal/cli/opsmanager/clusters/delete.go

```diff
@@ -43,39 +48,162 @@ func (opts *DeleteOpts) Run() error {
 	if !opts.Confirm {
 		return nil
 	}
+
+	err := opts.newHostIds()
+	if err != nil {
+		return err
+	}
+
+	// shutdown cluster
+	err = opts.shutdownCluster()
+	if err != nil {
+		return err
+	}
+
+	// Remove cluster from automation
+	err = opts.removeClusterFromAutomation()
+	if err != nil {
+		return err
+	}
+
+	// Stop monitoring
+	err = opts.stopMonitoring()
+	if err != nil {
+		return err
+	}
+
+	fmt.Print("Cluster deleted\n")
+	return nil
+}
+
+func (opts *DeleteOpts) newHostIds() error {
 	current, err := opts.store.GetAutomationConfig(opts.ConfigProjectID())
+	if err != nil {
+		return err
+	}
+	r := convert.FromAutomationConfig(current)
+	hostname := make(map[string][]int32)
+
+	for _, rs := range r {
+		if rs.Name == opts.Entry {
+			for _, s := range rs.ProcessConfigs {
+				hostname[s.Hostname] = append(hostname[s.Hostname], int32(s.Port))
+			}
+		}
+	}
 
+	hosts, err := opts.store.Hosts(opts.ConfigProjectID(), nil)
```

Comment:
we may get pagination problems here, ie, what if the host you're looking for is in the second page?, two options here, try to pass the max value allowed for items per page, or use 
[get-one-host-by-hostname-port](https://docs.opsmanager.mongodb.com/current/reference/api/hosts/get-one-host-by-hostname-port/)
Given you recently collected all processes and should have the hostname and port info, this may do more network calls but it may be the safest option in case of pagination edge cases